### ISSUE TITLE
Wrong operator for comparisons

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#334]
The compiler gets confused about the type of operators:
```
ergo$ return 1.0 < 2.0;
[RuntimeError at line 1 col 7] Binary operation failed.
return 1.0 < 2.0;
       ^^^^^^^^^ 
ergo$ return 1 < 2;
Response. true
```

This likely does not show up on the JavaScript backend due to a compensating bug in the code-generation, or the fact that both integers and doubles gets turned into numbers.

**Comment by jeromesimeon** [comment]
FYI @kach

**Comment by kach** [comment]
I still think the right answer here is to have a single numeric type. This would also solve the `+`/`+i` issue.

**Comment by jeromesimeon** [comment]
This is orthogonal to the number of numeric types we support. If we treat Integer as Double, both examples fail.

**Comment by jeromesimeon** [comment]
Fixed in PR #343

**Comment by jeromesimeon** [comment]
```
Welcome to ERGOTOP version 0.1.3
ergo$ return 1.0 < 2.0;
Response. true
ergo$ return 1 <i 2;
Response. true
```
